src/api/v1/router.ts

```diff
@@ -143,3 +150,38 @@ router.post('/service/wallet/import', async (req: Request, res: Response) => {
     return res.status(500).send({ error });
   }
 });
+
+router.post('/transaction/sign', async (req: Request, res: Response) => {
+  try {
+    const { accountId, transaction, encryptedWallet } = req.body;
+
+    const decryptedWallet: IDecryptedWalletService = await decryptWalletForService({ encryptedWallet });
+
+    if (!(decryptedWallet.isServiceAccount && accountId === decryptedWallet.accountId)) {
+      return res.status(400).send({
+        error: {
+          message: 'Not valid account',
+          code: ERROR_CODES.BAD_REQUEST
+        }
+      });
+    }
+    const unsignedTx = ethers.Transaction.from(transaction);
+    const signerWallet = new ethers.Wallet(decryptedWallet.privateKey);
+    const signedTransaction = await signerWallet.signTransaction(unsignedTx);
+
+    const data: IResponseSignedTransaction = {
+      signedTransaction
+    };
```

Comment:
Same as above - remember that cages do not allow outbound network requests (this is how we have configured them, for security) - so any values that are looked up from a network such as gas amount, gas price, nonce, etc. need to be passed into the cage in the API request.

Reply:
That is why it should be serialized transaction, it could contains different amount of properties.

Reply:
ok good :)